server/helpers/utils.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * @description Perform an async operation on array items
+ *
+ * @param {array} array array to iterate
+ * @param {function} callback a callback function
+ * @returns {Promise} a promise that resolves
+ * @method mapAsync
+ */
+export const mapAsync = async (array, callback) => {
+  const result = [];
+  const LENGTH = array.length;
```

Comment:
I have told you not to create a new variable for stuff like this. You are wasting memory.